bindings/rust/wasmedge-sys/src/vm.rs

```diff
@@ -0,0 +1,174 @@
+use super::wasmedge;
+use crate::{
+    raw_result::{decode_result, ErrReport},
+    string::StringRef,
+    value::Value,
+    wasi,
+};
+use std::os::raw::c_char;
+
+// If there is a third-party sdk based on wasmedge-sys
+// the private encapsulation  here can force the third-party sdk to use the api
+#[derive(Debug)]
+pub struct Vm {
+    pub(crate) ctx: *mut wasmedge::WasmEdge_VMContext,
+    _private: (),
```

Comment:
Please help not to show the private assets at this moment, because we do not have any. :)

Reply:
actually, based on the reply to my comment in the original PR, I think what you actually want is `#[non_exhaustive]` on structs with all `pub` members. This one has a `pub(crate)` in it, so it's already not constructible from outside of this crate.

However, this evidences that Clippy isn't being run. I _strongly_ suggest running `cargo clippy --all-features --profile test` on the code before shipping. Clippy is a great tool for learning Rust, too.

Reply:
clippy I use it a lot and it doesn't give you advice on everything. There is no advice or warning here either.

Reply:
I'm not trying to be `non_exhaustive` either, the field here was previously `pub` and I forgot to remove `_private`.